pallets/dapp-staking-v3/src/lib.rs

```diff
@@ -1603,6 +1584,57 @@ pub mod pallet {
 
             Ok(())
         }
+
+        /// A call used to fix accounts with inconsistent state, where frozen balance is actually higher than what's available.
+        ///
+        /// The approach is as simple as possible:
+        /// 1. Caller provides an account to fix.
+        /// 2. If account is eligible for the fix, all unlocking chunks are modified to be claimable immediately.
+        /// 3. The `claim_unlocked` call is executed using the provided account as the origin.
+        /// 4. All states are updated accordingly, and the account is no longer in an inconsistent state.
+        ///
+        /// The benchmarked weight of the `claim_unlocked` call is used as a base, and additional overestimated weight is added.
+        /// Call doesn't touch any storage items that aren't already touched by the `claim_unlocked` call, hence the simplified approach.
+        #[pallet::call_index(100)]
+        #[pallet::weight(T::WeightInfo::claim_unlocked(T::MaxNumberOfStakedContracts::get()).saturating_add(Weight::from_parts(1_000_000_000, 0)))]
+        pub fn fix_account(
+            origin: OriginFor<T>,
+            account: T::AccountId,
+        ) -> DispatchResultWithPostInfo {
+            Self::ensure_pallet_enabled()?;
+            ensure_signed(origin)?;
+
+            let mut ledger = Ledger::<T>::get(&account);
+            let locked_amount_ledger = ledger.total_locked_amount();
+            let total_balance = T::Currency::total_balance(&account);
+
+            if locked_amount_ledger > total_balance {
+                // 1. Modify all unlocking chunks so they can be unlocked immediately.
+                let current_block: BlockNumber =
+                    frame_system::Pallet::<T>::block_number().saturated_into();
+                ledger
+                    .unlocking
+                    .iter_mut()
+                    .for_each(|chunk| chunk.unlock_block = current_block);
+                Ledger::<T>::insert(&account, ledger);
+
+                // 2. Execute the unlock call, clearing all of the unlocking chunks.
+                let result = Self::internal_claim_unlocked(account);
+
+                // 3. Adjust consumed weight to consume the max possible weight (as defined in the weight macro).
```

Comment:
```suggestion
                // 3. Adjust consumed weight to consume the max possible weight (as defined in `T::WeightInfo::claim_unlocked()`).
```

nit: `internal_claim_unlocked()` has no weights macro, but use weights from `claim_unlocked()`.
I was confused at first

Reply:
I mean the macro above the call: `T::WeightInfo::claim_unlocked() + const.`